pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -288,9 +289,21 @@ func handleSyncOp(ctx *downloaderContext, key string,
 			}
 			continue
 		}
+
+		// Finalize: fsync and atomically move temp -> final.
+		if f, ferr := os.OpenFile(tempLocFilename, os.O_RDWR, 0); ferr == nil {
+			_ = f.Sync()
+			_ = f.Close()
+		}
+		if errStr := os.Rename(tempLocFilename, locFilename); errStr != nil {
+			log.Errorf("Rename failed: %v", errStr)
+			return handleSyncOpResponse(ctx, config, status, locFilename,
+				key, errStr.Error(), cancelled, cleanOnError)
+		}
+
 		// Record how much we downloaded
 		size := int64(0)
-		info, err := os.Stat(locFilename)
+		info, err := os.Stat(tempLocFilename)
```

Comment:
This must be wrong since you already did the rename hence tempLocFilename no longer exists.